app/views/catalog/_index_header_default.html.erb

```diff
@@ -1,6 +1,6 @@
 <div>
   <div class="video-container">
-    <video src="<%= document.proxy_src %>" poster="<%= document.thumb_src %>" 
+    <video src="<%= @ids_to_cached_proxy[document.id] %>" poster="<%= document.thumb_src %>" 
            loop muted oncontextmenu="return false;" preload="none"
```

Comment:
@mccalluc can we just override `document.proxy_src` in the controller, to keep the views unaware of caching machinery?